### PR TITLE
ENG 119 restructure json rpc api

### PR DESCRIPTION
**Pull request by danburck**
## Description

This PR reorganizes the  JSON-RPC apis and backend. 

Right now the changes include an example for the following proposal: 

- use `namespace/ethereum/<namespace>/api.go` files for an overview of exposed apis 
  - order by groups 
  
- use `backend` for api logic. Separate into several backend logic files:
  - Blocks
	- Read Transactions
	- Write Transactions
	- Account Information
	- EVM/Smart Contract Execution
	- Event Logs
	- Chain Information
	- Getting Uncles
	- Proof of Work
	- Other

### CODE REVIEW

**Comment by facs95** [comment]
Updated the PR doing the following changes:
- Merge the latest from main (with the new integration tests)
- Refractor main apis moving every clientquery to the backend

Still missing
- Reorganize `evm_backend` into different files

I will not refractor filters until we have an integration tests for them to make sure no regressions happen as the refractor is a bit more complex.

File: rpc/namespaces/ethereum/eth/api.go [hunk 1]
```diff
@@ -2,198 +2,312 @@ package eth
...
+///                           Chain Information													          ///
+///////////////////////////////////////////////////////////////////////////////
+
+// ProtocolVersion returns the supported Ethereum protocol version.
+func (e *PublicAPI) ProtocolVersion() hexutil.Uint {
+	e.logger.Debug("eth_protocolVersion")
+	return hexutil.Uint(ethermint.ProtocolVersion)
 }
 
 // GasPrice returns the current gas price based on Ethermint's gas price oracle.
```
Comment (danburck) [inline rpc/namespaces/ethereum/eth/api.go:313]: @facs95 Any reason, why you didn't move this endpoint logic?

**Review by danburck — COMMENTED** [review]
@facs95 Looks good, since it's already merged I think we should address the following things in future PRs:

* move remaining logic from eth/api.go (`GetTransactionByBlockNumberAndIndex()`, `GasPrice()`, `MaxPriorityFeePerGas()`,  `GetTransactionLogs()`, `GetPendingTransactions()`)
* clean up interfaces. We probably don't need both interfaces `EVMBackend` and `EthereumAPI` interface. What do you think?

File: rpc/namespaces/ethereum/eth/api.go (hunk 1 above)
Comment (facs95) [reply rpc/namespaces/ethereum/eth/api.go:313]: any method of `PublicAPI` gets exposed as an endpoint (`eth_protocolVersion`) its not the same when its a backend method. Those are not exposed. So I just moved the logic that use clientQuery or clientCtx to the backend. Anything that required state query.

**Pull request merged**